cleaner/cleaner.go

```diff
@@ -0,0 +1,38 @@
+package cleaner
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+type t func() error
+
+var cleaners []t
+var inactive bool
+
+// Register a cleaner function. When a function is registered, the Signal watcher is started in a goroutine.
+func Register(f t) {
```

Comment:
Would be neat to have a [variadic](https://gobyexample.com/variadic-functions) here, so we can register multiple cleaner easily. And now that I think about it, the cleaners should be executed in the reverse order they have been added, to handle possible dependency between them (as `defer` in go does).

Reply:
Done ! This was my first time using this concept in go ! Thanks !
It also inpired me to test the register function without uing the `os.Exit` part.

Reply:
As far as I can tell, you are not reversing the order. A simple way to fix that would be to prepend cleaners instead of appending them. See https://codingair.wordpress.com/2014/07/18/go-appendprepend-item-into-slice/

Reply:
I missed that part in your comment. Doing it now.